contracts/contracts/oracle/PriceFeedPair.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { AggregatorV3Interface } from "../interfaces/chainlink/AggregatorV3Interface.sol";
+import { StableMath } from "../utils/StableMath.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+/**
+ * @notice Price feed when the feed of 2 oracle prices need combining to achieve desired result.
+ *
+ * @dev multiplying oracle pair prices has combining properties. E.g. price FXS/USD multiplied by
+ * USD/ETH results effectively in FXS/ETH price. Since oracle prices express asset on the left priced
+ * by the asset on the right, we sometimes need to reverse prices in order to achieve desired results.
+ * ETH/USD reversed is USD/ETH.
+ *
+ * In our first usage of this contract we required FXS/ETH price. It can be derived using FXS/USD and
+ * ETH/USD prices. Since we need the latter reversed to get the desired result we configure the contract
+ * by using FXS/USD as feed 0 and USD/ETH (reversed from ETH/USD) as feed 1.
+ */
+contract PriceFeedPair is AggregatorV3Interface {
+    using SafeCast for uint256;
+    using SafeCast for int256;
+    using StableMath for uint256;
+
+    // Fields to make it compatible with `AggregatorV3Interface`
+    uint8 public constant override decimals = 18;
+    string public constant override description = "";
+    uint256 public constant override version = 1;
+    address public immutable addressFeed0;
+    address public immutable addressFeed1;
+    bool public immutable reverseFeed0;
+    bool public immutable reverseFeed1;
+    uint8 internal immutable decimalsFeed0;
+    uint8 internal immutable decimalsFeed1;
+
+    error PriceFeedAddressError(address _address);
+
+    constructor(
+        address _addressFeed0,
+        address _addressFeed1,
+        bool _reverseFeed0,
+        bool _reverseFeed1
+    ) {
+        if (_addressFeed0 == address(0)) {
```

Comment:
A check to ensure `_addressFeed0 != _addressFeed1` would be nice to have here

Reply:
Sure I can add that in.

I think we are going to need to fork test each feed configuration anyway, since combining 2 feeds and making it possible to reverse any of them leaves much room for error.